arch/posix/include/posix_cheats.h

```diff
@@ -30,7 +30,11 @@
 #define pthread_condattr_t     zap_pthread_condattr_t
 #define pthread_barrier_t      zap_pthread_barrier_t
 #define pthread_barrierattr_t  zap_pthread_barrierattr_t
+#define pthread_attr_t         zap_pthread_attr_t
```

Comment:
why is this #define instead of a typedef?
why do we have this zap_ layer of abstraction?

Reply:
Hi  @andrewboie, 

Sorry for my limited knowledge on this. I just followed the last followed the earlier one like pthread_barrierattr_t. It was suggested by @aescolar-ot to put it this way.

_zap abstraction layer was put to avoid the conflict of POSIX layer calls with native_posix architecture.

Reply:
Just a comment: This posix_cheats.h header, and its defines, are meant to rename any Zephyr global symbol (function or variable) whose name conflicts with the native/host POSIX API, so that they will have a different name when compiling for ARCH_POSIX.

Regarding types, we just need to ensure the headers, which include types conflicting with the POSIX API types, are not included by mistake from anything in arch/posix/ or boards/posix/ (or any of its drivers which may rely on talking with the real host APIs). So, we should not have these types included in any basic kernel header which ends up included almost everywhere else.